src/link/MachO/commands.zig

```diff
@@ -0,0 +1,378 @@
+const std = @import("std");
+const fs = std.fs;
+const io = std.io;
+const mem = std.mem;
+const macho = std.macho;
+const testing = std.testing;
+
+const Allocator = std.mem.Allocator;
+const makeName = @import("../MachO.zig").makeStaticString;
+
+pub const LoadCommand = union(enum) {
+    Segment: SegmentCommand,
+    DyldInfoOnly: macho.dyld_info_command,
+    Symtab: macho.symtab_command,
+    Dysymtab: macho.dysymtab_command,
+    Dylinker: GenericCommandWithData(macho.dylinker_command),
+    Dylib: GenericCommandWithData(macho.dylib_command),
+    Main: macho.entry_point_command,
+    VersionMin: macho.version_min_command,
+    SourceVersion: macho.source_version_command,
+    LinkeditData: macho.linkedit_data_command,
+    Unknown: GenericCommandWithData(macho.load_command),
+
+    pub fn read(allocator: *Allocator, reader: anytype) !LoadCommand {
+        const header = try reader.readStruct(macho.load_command);
+        var buffer = try allocator.alloc(u8, header.cmdsize);
+        defer allocator.free(buffer);
+        const slice = [1]macho.load_command{header};
+        mem.copy(u8, buffer[0..], mem.sliceAsBytes(slice[0..1]));
```

Comment:
Why not `mem.asBytes(&header)` or `mem.toBytes(header)` ?
Ditto for all the other uses of `sliceAsBytes` with single-item slices.

Reply:
Yeah, good point. I didn't use it as I was following the convention set out by `Elf` linker. This is not to say this cannot be refactored out to use `mem.asBytes` though. Having said that, I'd leave it out of the scope of this PR and have it cleaned up in a subsequent PR if you don't mind?

Reply:
Not at all, I was just wondering why there were so many single-item slices.

Reply:
So I've not changed every occurrence in the MachO linker yet, but while we're here, I've changed all uses of `mem.sliceAsBytes` to `mem.asBytes` in `commands.zig`, and in a subsequent PR I can refactor the rest elsewhere in the linker.